### DISCUSSION TITLE
Can PhysX Visual Debugger be used with PhysX 5 + Running SnippetOmniPVD throws exception

### DISCUSSION THREAD

**Discussion by itamar-dw** [NVIDIA-Omniverse/PhysX#81] [Q&A]
I have 2 questions which might be related so I am putting both in a single message:

1. I understand that the OmniPVD can be used, but if I don't have Omniverse installed, can I use the visual debugger from here https://developer.nvidia.com/physx-visual-debugger? Can I refer to its manual or is it outdated with PhysX 5?
2. I am trying to run the SnippetOmniPVD snippet, I thought maybe the output file can be loaded with the PhysX Visual Debugger mentioned above. However, code fails with exception, the reason being that `gOmniPVD` variable in `SnippetOmniPVD.cpp` https://github.com/NVIDIA-Omniverse/PhysX/blob/4d33492253ab64c2246d80257ac4ce0f37b40b22/physx/snippets/snippetomnipvd/SnippetOmniPvd.cpp#L86 gets a null value and the method `initPhysicsWithOmniPvd` returns without creating `gPhysics`. Any idea what goes wrong?

**Comment by itamar-dw** [comment]
Ok, I figured it out, turns out the CMake flag to create the `PVDRuntime` project was `False` by default. I switched it to `True` and built the project, and now there is no exception and I get the output PVD file.

However, I still want to know how can I use the visual debugger (either from file or by streaming), if I don't have the Omniverse installed.

**Reply by pdherasciec** [reply]
To answer about running OmniPVD (the visual inspection part) without Omniverse; for the forseeable future Omniverse will have to be installed to debug OVD files.

**Comment by pdherasciec** [comment]
Hello, is it possible that the SnippetOmniPVD was run in Release mode? PhysX is made not to ship with OmniPVD output in Release as OmniPVD incurs some overhead/book keeping and Release is supposed to be as lean and fast as possible.

**Reply by itamar-dw** [reply]
> Hello, is it possible that the SnippetOmniPVD was run in Release mode? PhysX is made not to ship with OmniPVD output in Release as OmniPVD incurs some overhead/book keeping and Release is supposed to be as lean and fast as possible.

No, it happens in Debug mode.
Btw to solve it I didn't change the Cmake file directly, I just added this line to the xml preset (in `physx\buildtools\presets\public` folder):

`<cmakeSwitch name="PX_BUILDPVDRUNTIME" value="True" comment="Build the PVD runtime" />`

**Reply by pdherasciec** [reply]
Oh that's curious, that's not supposed to be the case, will have to look into that. Is it Linux or Windows?

**Reply by pdherasciec** [reply]
Actually would you please say which xml file it is you edited like that?

**Reply by itamar-dw** [reply]
> Oh that's curious, that's not supposed to be the case, will have to look into that. Is it Linux or Windows?

Windows with Visual Studio 2022

> Actually would you please say which xml file it is you edited like that?

Sorry, you are right. I based my build on this commit for supporting VS 2022, and for some reason it omits the cmake switch for building PVD runtime. I forgot I used an external xml (not from the PhysX repo)...
https://github.com/NVIDIAGameWorks/PhysX/commit/00180480a25c8ac6172d3703a4939af8dc06812b

**Reply by pdherasciec** [reply]
Good so everything is in order. I was afraid the xml files got somehow mangled in our public release. Thanks for following up!

**Reply by MattiaMontanari** [reply]
@itamar-dw , did you manage to use PVD with PhysX? Did you have to install and use the newer OVD to profile?

**Reply by itamar-dw** [reply]
> @itamar-dw , did you manage to use PVD with PhysX? Did you have to install and use the newer OVD to profile?

No, I wasn't. I don't have a GPU with the RTX feature that is needed for Omniverse, which is needed for PVD (I tried but apps kept crashing).
